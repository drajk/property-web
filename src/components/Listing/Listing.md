# Listing.js

Shows a styled Listing with action.

## Usage

```html
<Listing price="980000" address="12 Araleun Drive, Hawthorn" icon="star" />
<Listing price="980000" address="12 Araleun Drive, Hawthorn" icon="remove" />
<Listing price="980000" address="12 Araleun Drive, Hawthorn" icon="remove" onClick={() => {}}/>
```
