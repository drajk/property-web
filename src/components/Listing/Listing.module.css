.wrapper {
    display: flex;
    flex-direction: row;
    background-color: var(--white);
    margin-bottom: 20px;
    border-radius: 4px;
    padding: 20px;
    box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.15);
    font-size: var(--font-size-small);
    min-width: 300px;
}

.content {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    justify-content: space-between;
}

.icon {
    padding: var(--padding);
    text-align: center;
    height: 30px;
}
