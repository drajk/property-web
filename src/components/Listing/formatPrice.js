const formatPrice = (value) => {
    const formatter = new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'AUD',
        minimumFractionDigits: 0,
    });

    return formatter.format(value);
};

export default formatPrice;
