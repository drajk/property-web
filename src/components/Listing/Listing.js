import React from 'react';
import PropTypes from 'prop-types';
import formatPrice from './formatPrice';
import star from './svg/star.svg';
import remove from './svg/remove.svg';

import styles from './Listing.module.css';

const Listing = ({ price, address, icon }) => {

    // TODO: move this to a Icon component
    const svgAsset = icon === 'star' ? star : remove;
    
    return (
        <div className={styles.wrapper}>
            <div className={styles.content}>
                <span>{formatPrice(price)}</span>
                <span>{address}</span>
            </div>
            <div className={styles.action}>
                <img className={styles.icon} src={svgAsset} alt={icon} />
            </div>
        </div>
    );
};

Listing.defaultProps = {
    price: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    icon: PropTypes.oneOf(['star', 'remove']),
};

export default Listing;
