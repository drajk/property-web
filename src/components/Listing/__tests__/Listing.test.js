import React from 'react';
import renderer from 'react-test-renderer';
import Listing from '../Listing';

describe('Listing', () => {
    it('renders with formatted price and address', () => {
        const tree = renderer
            .create(
                <Listing
                    price="980000"
                    address="12 Araleun Drive, Hawthorn"
                    icon="star"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with remove favourite icon', () => {
        const tree = renderer
            .create(
                <Listing
                    price="980000"
                    address="12 Araleun Drive, Hawthorn"
                    icon="star"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with remove icon', () => {
        const tree = renderer
            .create(
                <Listing
                    price="980000"
                    address="12 Araleun Drive, Hawthorn"
                    icon="remove"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with on click handler', () => {
        const tree = renderer
            .create(
                <Listing
                    price="980000"
                    address="12 Araleun Drive, Hawthorn"
                    icon="remove"
                    onClick={() => {}}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
