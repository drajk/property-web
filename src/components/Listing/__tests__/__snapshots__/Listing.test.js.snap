// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing renders with formatted price and address 1`] = `
<div
  className="wrapper"
  data-testid="listing-wrapper"
>
  <div
    className="content"
  >
    <span
      data-testid="listing-price"
    >
      A$980,000
    </span>
    <span
      data-testid="listing-address"
    >
      12 Araleun Drive, Hawthorn
    </span>
  </div>
  <div
    className="action"
  >
    <img
      alt="star"
      className="icon"
      data-testid="listing-icon"
      src="star.svg"
    />
  </div>
</div>
`;

exports[`Listing renders with on click handler 1`] = `
<div
  className="wrapper"
  data-testid="listing-wrapper"
>
  <div
    className="content"
  >
    <span
      data-testid="listing-price"
    >
      A$980,000
    </span>
    <span
      data-testid="listing-address"
    >
      12 Araleun Drive, Hawthorn
    </span>
  </div>
  <div
    className="action"
  >
    <img
      alt="remove"
      className="icon"
      data-testid="listing-icon"
      src="remove.svg"
    />
  </div>
</div>
`;

exports[`Listing renders with remove favourite icon 1`] = `
<div
  className="wrapper"
  data-testid="listing-wrapper"
>
  <div
    className="content"
  >
    <span
      data-testid="listing-price"
    >
      A$980,000
    </span>
    <span
      data-testid="listing-address"
    >
      12 Araleun Drive, Hawthorn
    </span>
  </div>
  <div
    className="action"
  >
    <img
      alt="star"
      className="icon"
      data-testid="listing-icon"
      src="star.svg"
    />
  </div>
</div>
`;

exports[`Listing renders with remove icon 1`] = `
<div
  className="wrapper"
  data-testid="listing-wrapper"
>
  <div
    className="content"
  >
    <span
      data-testid="listing-price"
    >
      A$980,000
    </span>
    <span
      data-testid="listing-address"
    >
      12 Araleun Drive, Hawthorn
    </span>
  </div>
  <div
    className="action"
  >
    <img
      alt="remove"
      className="icon"
      data-testid="listing-icon"
      src="remove.svg"
    />
  </div>
</div>
`;
